pkg/user/tx_client.go

```diff
@@ -127,6 +129,15 @@ func WithDefaultAccount(name string) Option {
 	}
 }
 
+// WithEstimatorService allows a user to provide a connection to a special gas
+// estimation service to be used by the TxClient for estimating gas price
+// and usage.
+func WithEstimatorService(conn *grpc.ClientConn) Option {
```

Comment:
FOR CALLUM: 

Please note that when this service is set, we *exclusively* use this estimator service for estimation -- if it fails, there is no fallback of using the default connection (not currently implemented in this PR). 

If a user configures a separate estimator service to use, I think the failure of the service should be explicit + noticeable to the user. 

Thoughts?